Race condition that fails a query

Stack trace:

```
Caused by: java.lang.IllegalStateException: Expected a single output buffer for task 20140918_000254_00010_a2utt.0.0, but found []
    at com.google.common.base.Preconditions.checkState(Preconditions.java:200)
    at com.facebook.presto.server.StatementResource$Query.updateExchangeClient(StatementResource.java:431)
    at com.facebook.presto.server.StatementResource$Query.getData(StatementResource.java:395)
    at com.facebook.presto.server.StatementResource$Query.getNextResults(StatementResource.java:314)
    at com.facebook.presto.server.StatementResource$Query.getResults(StatementResource.java:308)
    at com.facebook.presto.server.StatementResource.getQueryResults(StatementResource.java:221)
    at sun.reflect.GeneratedMethodAccessor521.invoke(Unknown Source)
```

This happens fairly regularly in the CI (same issue, but different test run from the above):

```
com.facebook.presto.client.FailureInfo$FailureException: Expected response code from http://127.0.0.1:43251/v1/task/20140918_231251_00351_xikpe.1.3?summarize to be 200, but was 500: Request failed.
    at com.facebook.presto.server.HttpRemoteTask$SimpleHttpResponseHandler.onSuccess(HttpRemoteTask.java:748)
    at com.facebook.presto.server.HttpRemoteTask$SimpleHttpResponseHandler.onSuccess(HttpRemoteTask.java:717)
    at com.google.common.util.concurrent.Futures$4.run(Futures.java:1181)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
    at java.lang.Thread.run(Thread.java:722)
```

--- a/presto-main/src/main/java/com/facebook/presto/server/StatementResource.java
+++ b/presto-main/src/main/java/com/facebook/presto/server/StatementResource.java
@@ -428,6 +428,10 @@ public class StatementResource
             if (!outputStage.getState().isDone()) {
                 for (TaskInfo taskInfo : outputStage.getTasks()) {
                     List<BufferInfo> buffers = taskInfo.getOutputBuffers().getBuffers();
+                    if (buffers.isEmpty()) {
+                        // output buffer has not been created yet
+                        continue;
+                    }
                     Preconditions.checkState(buffers.size() == 1,
                             "Expected a single output buffer for task %s, but found %s",
                             taskInfo.getTaskId(),

